src/score_hv/harvesters/global_bucket_precip_ave.py

```diff
@@ -0,0 +1,129 @@
+                       "variable to harvest from the background forecast data. "
+                       "Please reconfigure the input dictionary using only the "
+                       "following variables: %r" % (var, VALID_VARIABLES))
+                raise KeyError(msg)
+    
+    def get_stats(self):
+        ''' return list of all stat types based on harvest_config '''
+        return self.stats
+
+    def set_stats(self):
+        """
+        set the statistics specified by the config dict
+        """
+        self.stats = self.config_data.get('statistic')
+        for stat in self.stats:
+            if stat not in VALID_STATISTICS:
+                msg = ("'%s' is not a supported statistic to harvest for global bucket precipitation. "
+                       "Please reconfigure the input dictionary using only the "
+                       "following statistics: %r" % (stat, VALID_STATISTICS))
+                raise KeyError(msg)
+    
+
+
+    def get_variables(self):
+        ''' return list of all variable types based on harvest_config'''
+        return self.variables
+
+@dataclass
+
+
+@dataclass
+class GlobalBucketPrecipRateHv(object):
+    """ Harvester dataclass used to parse precip stored in 
+        background forecast data
+    
+        Parameters:
+        -----------
+        config: TemperatureConfig object containing information used to determine
+                which variable to parse the log file
+        Methods:
+        --------
+        get_data: parse descriptive statistics from log files based on input
+                  config file
+    
+                  returns a list of tuples containing specific data
+    """
+    config: GlobalBucketPrecipRateConfig = field(
+                                default_factory=GlobalBucketPrecipRateConfig)
+    
+    def get_data(self):
+        """ Harvests prateb_ave from background forecast data
+            returns harvested_data, a list of HarvestData tuples
+        """
+        harvested_data = list()
+       
+        mean_values = []
+        filenames   = self.config.harvest_filenames
+        #Open the requested file names
+        for infile in filenames:
+            nc_file = nc.Dataset(infile, 'r')
+        #Open the requested file names
+        for i, variable in enumerate(self.config.get_variables()):
+            """ The first nested loop iterates through each requested variable
+            """
+            for j, statistic in enumerate(self.config.get_stats()):
+                """ The second nested loop iterates through each requested 
+                    statistic
+                """
+                var_name    = variable
+                for infile in filenames:
+                    nc_file            = nc.Dataset(infile, 'r')
+                    requested_var      = nc_file.variables[var_name][:]
+                    units              = nc_file.variables[var_name].units if hasattr(nc_file.variables[var_name], 'units') else None
```

Comment:
line 119 is compact and it works. But stylistically, I would prefer:
```suggestion
                    if hasattr(nc_file.variables[var_name], 'units'):
                        units = nc_file.variables[var_name].units
                    else:
                        units = None
```

Reply:
seconding the style recommendation here